oak_functions/loader/src/main.rs

```diff
@@ -48,48 +53,57 @@ pub struct Opt {
         help = "Path to a Wasm file to be loaded and executed per invocation. The Wasm module must export a function named `main`."
     )]
     wasm_path: String,
-    // TODO(#1930): Support downloading lookup data from a URL instead of a local file.
     // TODO(#1930): Support periodically re-downloading lookup data at a given time interval.
     #[structopt(
         long,
-        help = "Path to a file to load for data lookups, containing key / value entries in protobuf binary format."
+        help = "URL to a file to GET over HTTP containing key / value entries in protobuf binary format for lookup."
     )]
-    lookup_data_path: String,
+    lookup_data_url: String,
 }
 
 struct LookupData {
-    lookup_data_path: String,
+    lookup_data_url: String,
     entries: RwLock<HashMap<Vec<u8>, Vec<u8>>>,
 }
 
 impl LookupData {
     /// Creates a new [`LookupData`] instance that can refresh its internal entries from the
-    /// provided data file path.
+    /// provided data file URL.
     ///
     /// Entries in the data file path must be consecutive binary encoded and length delimited
     /// protobuf messages according to the definition in `/oak_functions/proto/lookup_data.proto`.
     ///
     /// The returned instance is empty, and must be populated by calling the [`LookupData::refresh`]
     /// method at least once.
-    fn new_empty(lookup_data_path: &str) -> LookupData {
+    fn new_empty(lookup_data_url: &str) -> LookupData {
         LookupData {
-            lookup_data_path: lookup_data_path.to_string(),
+            lookup_data_url: lookup_data_url.to_string(),
             entries: RwLock::new(HashMap::new()),
         }
     }
 
-    /// Refreshes the internal entries of this struct from the data file path provided at
+    /// Refreshes the internal entries of this struct from the data file URL provided at
     /// construction time.
     ///
     /// If successful, entries are completely replaced (i.e. not merged).
     ///
     /// If there is any error while reading or parsing the data, an error is returned by this
     /// method, and existing entries are left untouched. The caller may retry the refresh operation
     /// at a future time.
-    fn refresh(&self) -> anyhow::Result<()> {
+    async fn refresh(&self) -> anyhow::Result<()> {
         // TODO(#1930): Avoid loading the entire file in memory for parsing.
-        let lookup_data_buf =
-            fs::read(&self.lookup_data_path).context("could not read lookup data file")?;
+        let client = hyper::Client::new();
+        let res = client
+            .get(
+                self.lookup_data_url
+                    .parse()
+                    .context("could not parse lookup data URL")?,
+            )
+            .await
+            .context("could not fetch lookup data")?;
+        let lookup_data_buf = hyper::body::to_bytes(res.into_body())
```

Comment:
Out of curiosity: We use length-delimited protobufs to be able to read and parse large files in chunks. Does `hyper::Client` allow reading a file in chunks?

Reply:
AFAICT hyper allows reading a request body in a streaming fashion, yes, I left a TODO somewhere to not buffer the entire request in memory, but I'm not sure how easy it will be in practice.